core/server/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -114,28 +111,32 @@ public BlockWorkerClientServiceHandler getWorkerServiceHandler() {
    * @throws IOException if an IO exception occurs
    */
   public DefaultBlockWorker() throws IOException {
+    this(new BlockMasterClient(NetworkAddressUtils.getConnectAddress(ServiceType.MASTER_RPC)),
+        new FileSystemMasterClient(NetworkAddressUtils.getConnectAddress(ServiceType.MASTER_RPC)),
+        new Sessions(), new TieredBlockStore());
+  }
+
+  /**
+   * Constructs a default block worker.
+   *
+   * @param blockMasterClient a client for talking to the block master
+   * @param fileSystemMasterClient a client for talking to the file system master
+   * @param sessions an object for tracking and cleaning up client sessions
+   * @param blockStore an Alluxio block store
+   * @throws IOException if an IO exception occurs
+   */
+  public DefaultBlockWorker(BlockMasterClient blockMasterClient,
+      FileSystemMasterClient fileSystemMasterClient, Sessions sessions, BlockStore blockStore)
```

Comment:
It gets a little weird when the dependencies need `this` in their constructors. As it is, I'm treating the syncers/cleaners as part of the same unit as the `BlockWorker`, so they aren't injected. We may want to put some effort into rearchitecting the class at some point so that it does less things and has a more clearly defined relationship with its dependencies.